app/src/store/store.ts

```diff
@@ -43,19 +52,42 @@ const store: StoreOptions<RootState> = {
       const location = LocationService.findByTitle(state.locationSearchModule.selectedLocation);
       const searchValues = state.textSearchModule.searchValues;
       const radius = state.locationSearchModule.selectedRadius;
-      return new Promise((resolve, reject) => {
+
+      const from = state.resultsFrom;
+      const size = state.resultSetSize;
+
+      return new Promise((resolve) => {
         DataService.findBySelection({
           searchValues,
           location,
-          radius
-        }).then((result) => {
-          commit('setPosts', result);
-          resolve(result);
+          radius,
+          from,
+          size
+        }).then((result: PaginatedResponse<Post>) => {
+          commit('setTotalResultSize', result.meta.total);
+          commit('setPosts', result.data);
+          resolve(result.data);
         });
       });
 
     },
-    setPage({ commit, dispatch }, page: number): void {
+    setPage({ commit, dispatch, state }, page: number): void {
+        if (page < 1) {
+          page = 1;
+        }
+        // Calculate the new from parameter to load the next resultSet chunk if necessary
+        const currentPageIndex = (page - 1) * state.hitsPerPage; // these hu-mons start counting their pages at 1...
+        const currentLoadedChunk = {
+          min: state.resultsFrom,
+          max: state.resultsFrom + state.resultSetSize - 1 // again with these hu-mons and their count beginning at 1...
+        };
+        if (! inChunk(currentPageIndex, currentLoadedChunk)) {
+          // Calculate the needed offset
+          // rounding off to the next multiple of our resultSetSize
+          const from = currentPageIndex - (currentPageIndex % state.resultSetSize);
```

Comment:
Modulo is used at this point.

Reply:
Yes, to calculate the offset from where to load the posts from elastic (and this is working). 
We are looking for this one: https://github.com/hochschule-darmstadt/einander-helfen/blob/issue/88/app/src/views/ResultPage.vue#L278 where the same calculation should have happend.
This is a clear case of abstraction leaking and will be worked upon.